### ISSUE TITLE
Issue : Installation failed due to "npm: command not found"

### ISSUE THREAD

**Issue by kvaes** [Microsoft/openshift-origin#19]
[root@aacl-master-0 0]# cat stderr
warning: /var/cache/yum/x86_64/7/epel/packages/ansible-2.3.1.0-1.el7.noarch.rpm: Header V3 RSA/SHA256 Signature, key ID 352c64e5: NOKEY
Importing GPG key 0x352C64E5:
 Userid     : "Fedora EPEL (7) <epel@fedoraproject.org>"
 Fingerprint: 91e9 7d7c 4a5e 96f1 7f3e 888f 6a2f aea2 352c 64e5
 Package    : epel-release-7-10.noarch (installed)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
INFO: Volume group backing root filesystem could not be determined
File descriptor 10 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on pvs invocation. Parent PID 1760: /bin/bash
File descriptor 11 (/dev/urandom) leaked on pvs invocation. Parent PID 1760: /bin/bash
Checking that no-one is using this disk right now ...
OK
sfdisk:  /dev/sdc: unrecognized partition table type

sfdisk: No partitions found

Warning: partition 1 does not end at a cylinder boundary
Warning: no primary partition is marked bootable (active)
This does not matter for LILO, but the DOS MBR will not boot this disk.
If you created or changed a DOS partition, /dev/foo7, say, then use dd(1)
to zero the first 512 bytes:  dd if=/dev/zero of=/dev/foo7 bs=512 count=1
(See fdisk(8).)
INFO: Device node /dev/sdc1 exists.
File descriptor 10 (/dev/urandom) leaked on pvcreate invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on pvcreate invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on vgcreate invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgcreate invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on lvs invocation. Parent PID 1811: /bin/bash
File descriptor 11 (/dev/urandom) leaked on lvs invocation. Parent PID 1811: /bin/bash
File descriptor 10 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on vgs invocation. Parent PID 1819: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgs invocation. Parent PID 1819: /bin/bash
File descriptor 10 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on vgs invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on lvcreate invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on lvcreate invocation. Parent PID 1741: /bin/bash
File descriptor 10 (/dev/urandom) leaked on lvs invocation. Parent PID 1868: /bin/bash
File descriptor 11 (/dev/urandom) leaked on lvs invocation. Parent PID 1868: /bin/bash
File descriptor 10 (/dev/urandom) leaked on lvm invocation. Parent PID 1917: /bin/bash
File descriptor 11 (/dev/urandom) leaked on lvm invocation. Parent PID 1917: /bin/bash
File descriptor 10 (/dev/urandom) leaked on lvchange invocation. Parent PID 1741: /bin/bash
File descriptor 11 (/dev/urandom) leaked on lvchange invocation. Parent PID 1741: /bin/bash
Created symlink from /etc/systemd/system/multi-user.target.wants/docker.service to /usr/lib/systemd/system/docker.service.
Error: Package: 1:nodejs-6.11.1-1.el7.x86_64 (epel)
           Requires: libhttp_parser.so.2()(64bit)
Error: Package: 1:nodejs-6.11.1-1.el7.x86_64 (epel)
           Requires: http-parser >= 2.7.0
masterPrep.sh: line 77: npm: command not found

**Comment by kvaes** [comment]
su - 
yum install gcc-c++ openssl-devel
cd /usr/local/src
wget http://nodejs.org/dist/node-latest.tar.gz
tar zxvf node-latest.tar.gz
(cd into extracted folder: ex "cd node-v0.10.3")
./configure
make
make install

git clone https://github.com/isaacs/npm.git
cd npm
sudo make install

[root@aacl-master-0 npm]# cd /var/lib/waagent/custom-script/download/0/
[root@aacl-master-0 0]# ./masterPrep.sh


...


gyp WARN EACCES user "root" does not have permission to access the dev dir "/usr/local/lib/node_modules/azure-cli/node_modules/fibers/.node-gyp/8.4.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/azure-cli/node_modules/fibers/.node-gyp"
gyp WARN EACCES user "root" does not have permission to access the dev dir "/usr/local/lib/node_modules/azure-cli/node_modules/fibers/.node-gyp/8.4.0"
gyp WARN EACCES attempting to reinstall using temporary dev dir "/usr/local/lib/node_modules/azure-cli/node_modules/fibers/.node-gyp"

**Comment by BorisB2015** [comment]
I'm trying to replace Azure CLI 1.0 with Azure CLI 2.0. Once I get all pieces working/merged, I will close this issue.

[See here](https://github.com/BorisB2015/openshift-origin/commit/3511ac902e77b35d3d2fe6e33fa68a4cd90bb6d4#diff-5122bd37142550977e7f76c928502618)

**Comment by heatmiser** [comment]
In the meantime, until CentOS 7.4 is released, you can use the code from this repo, just add the following line to masterPrep.sh prior to installing nodejs:

rpm -ivh https://kojipkgs.fedoraproject.org//packages/http-parser/2.7.1/3.el7/x86_64/http-parser-2.7.1-3.el7.x86_64.rpm

**Comment by haroldwongms** [comment]
This is a duplicate of another issue.

We are removing the installation of the Azure CLI from the latest templates so this will go away.

Once Boris gets a chance to review and merge, this will no longer be an issue.

Thank you.

Harold

**Comment by haroldwongms** [comment]
The new changes are in effect so this should no longer be an issue.  Closing this.
